### DISCUSSION TITLE
new row violates row-level security policy for table

### DISCUSSION THREAD

**Discussion by rphly** [supabase/supabase#2108] [Questions]
Unable to insert row even though token is valid, and user has verified email.

![Screenshot 2021-06-27 at 11 27 39 PM](https://user-images.githubusercontent.com/23185489/123550394-cdf3a200-d79f-11eb-9752-29d2a9fd2012.png)

Receiving
`{'data': {'hint': None, 'details': None, 'code': '42501', 'message': 'new row violates row-level security policy for table "messages"'}, 'status_code': 403}`

**Comment by silentworks** [comment]
But is the user logged in? because the RLS for `INSERT` is checking if the user role is authenticated, you can have a user who has been verified but not logged in.

Can you share at what point of the process you are trying to do this insert also please?

**Reply by rphly** [reply]
Apologies for the lack of detail.

The user is logged in, there is an active session.

The error occurs on .execute() when I insert a row. Doesn't raise an exception, but it is returned through the response body.

I'm using supabase-py.

**Reply by silentworks** [reply]
I haven't written python in a long while, but I will try and set up a python project tomorrow and see if I get the same error. Or if you could create a repo without your supabase credentials in it and share here then I could try and debug from there.

**Reply by eMeRiKa13** [reply]
Did you find a solution?

I have the same problem. User auth but this error

**Reply by martypdx** [reply]
did you try also adding select permissions?
